From 6e4e3dd2b373b45ff2a29b65ed3b53657fb5c325 Mon Sep 17 00:00:00 2001
From: Xavier Boniface <xavierx.boniface@intel.com>
Date: Thu, 25 Apr 2013 18:46:12 +0200
Subject: [PORT FROM MAIN][BT] Firmware file different on ffrd and on vv

BZ: 103648

In bt_vendor_xxx.conf, the path we now set for the Bluetooth firmware files is
not the final path any more but a 'root' path containing a vv/ and a pr/
folder.
At run-time, depending on whether we are on a vv board or on a form factor,
firmwares will be seeked in the correct directory.

Notes:
 - for now, we rely on the ro.spid.wifi.nvram property to know if we are on
a vv or a ffrd. This will soon be abandoned by Wi-Fi. This property should be
renamed then.
 - For TI chips, the firmware file location is not set here

Change-Id: Ib21c8dc0698247f24fb56003b37f73a3659653af
Orig-Change-Id: Ic15985896b0d97e53da3a4ed960f1fec0694b39f
Feature: BT
Category: Integration
Purpose: ABSP-Adapter
Origin: Internal
Upstream-candidate: no
Signed-off-by: Xavier Boniface <xavierx.boniface@intel.com>
Reviewed-on: http://android.intel.com:8080/105036
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Aquilone, RaffaeleX <raffaelex.aquilone@intel.com>
Reviewed-by: Gros, Julien <julien.gros@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 libbt/conf/intel/baylake/bt_vendor.conf     |    2 +-
 libbt/conf/intel/bodegabay/bt_vendor.conf   |    2 +-
 libbt/conf/intel/ctpscaleht/bt_vendor.conf  |    2 +-
 libbt/conf/intel/ctpscalelt/bt_vendor.conf  |    2 +-
 libbt/conf/intel/redhookbay/bt_vendor.conf  |    3 +-
 libbt/conf/intel/saltbay_pr0/bt_vendor.conf |    2 +-
 libbt/conf/intel/saltbay_pr1/bt_vendor.conf |    2 +-
 libbt/conf/intel/victoriabay/bt_vendor.conf |    2 +-
 libbt/src/conf.c                            |    2 +
 libbt/src/hardware.c                        |   30 +++++++++++++++++++++++++++
 10 files changed, 40 insertions(+), 9 deletions(-)

diff --git a/libbt/conf/intel/baylake/bt_vendor.conf b/libbt/conf/intel/baylake/bt_vendor.conf
index 62ffeb9..93aa23c 100644
--- a/libbt/conf/intel/baylake/bt_vendor.conf
+++ b/libbt/conf/intel/baylake/bt_vendor.conf
@@ -2,4 +2,4 @@
 UartPort = /dev/ttyMFD0
 
 # Firmware patch file location
-FwPatchFilePath = /etc/firmware/
+FwPatchFileRootPath = /etc/firmware/bt/
diff --git a/libbt/conf/intel/bodegabay/bt_vendor.conf b/libbt/conf/intel/bodegabay/bt_vendor.conf
index 62ffeb9..93aa23c 100644
--- a/libbt/conf/intel/bodegabay/bt_vendor.conf
+++ b/libbt/conf/intel/bodegabay/bt_vendor.conf
@@ -2,4 +2,4 @@
 UartPort = /dev/ttyMFD0
 
 # Firmware patch file location
-FwPatchFilePath = /etc/firmware/
+FwPatchFileRootPath = /etc/firmware/bt/
diff --git a/libbt/conf/intel/ctpscaleht/bt_vendor.conf b/libbt/conf/intel/ctpscaleht/bt_vendor.conf
index 62ffeb9..93aa23c 100644
--- a/libbt/conf/intel/ctpscaleht/bt_vendor.conf
+++ b/libbt/conf/intel/ctpscaleht/bt_vendor.conf
@@ -2,4 +2,4 @@
 UartPort = /dev/ttyMFD0
 
 # Firmware patch file location
-FwPatchFilePath = /etc/firmware/
+FwPatchFileRootPath = /etc/firmware/bt/
diff --git a/libbt/conf/intel/ctpscalelt/bt_vendor.conf b/libbt/conf/intel/ctpscalelt/bt_vendor.conf
index 62ffeb9..93aa23c 100644
--- a/libbt/conf/intel/ctpscalelt/bt_vendor.conf
+++ b/libbt/conf/intel/ctpscalelt/bt_vendor.conf
@@ -2,4 +2,4 @@
 UartPort = /dev/ttyMFD0
 
 # Firmware patch file location
-FwPatchFilePath = /etc/firmware/
+FwPatchFileRootPath = /etc/firmware/bt/
diff --git a/libbt/conf/intel/redhookbay/bt_vendor.conf b/libbt/conf/intel/redhookbay/bt_vendor.conf
index 26a7e5f..6a5ebf3 100644
--- a/libbt/conf/intel/redhookbay/bt_vendor.conf
+++ b/libbt/conf/intel/redhookbay/bt_vendor.conf
@@ -1,5 +1,4 @@
 # UART device port where Bluetooth controller is attached
 UartPort = /dev/hci_tty
 
-# Firmware patch file location
-FwPatchFilePath = /etc/firmware/
+# Note: for TI chips, the firmware file location is not set here
diff --git a/libbt/conf/intel/saltbay_pr0/bt_vendor.conf b/libbt/conf/intel/saltbay_pr0/bt_vendor.conf
index 62ffeb9..93aa23c 100644
--- a/libbt/conf/intel/saltbay_pr0/bt_vendor.conf
+++ b/libbt/conf/intel/saltbay_pr0/bt_vendor.conf
@@ -2,4 +2,4 @@
 UartPort = /dev/ttyMFD0
 
 # Firmware patch file location
-FwPatchFilePath = /etc/firmware/
+FwPatchFileRootPath = /etc/firmware/bt/
diff --git a/libbt/conf/intel/saltbay_pr1/bt_vendor.conf b/libbt/conf/intel/saltbay_pr1/bt_vendor.conf
index 62ffeb9..93aa23c 100644
--- a/libbt/conf/intel/saltbay_pr1/bt_vendor.conf
+++ b/libbt/conf/intel/saltbay_pr1/bt_vendor.conf
@@ -2,4 +2,4 @@
 UartPort = /dev/ttyMFD0
 
 # Firmware patch file location
-FwPatchFilePath = /etc/firmware/
+FwPatchFileRootPath = /etc/firmware/bt/
diff --git a/libbt/conf/intel/victoriabay/bt_vendor.conf b/libbt/conf/intel/victoriabay/bt_vendor.conf
index 62ffeb9..93aa23c 100644
--- a/libbt/conf/intel/victoriabay/bt_vendor.conf
+++ b/libbt/conf/intel/victoriabay/bt_vendor.conf
@@ -2,4 +2,4 @@
 UartPort = /dev/ttyMFD0
 
 # Firmware patch file location
-FwPatchFilePath = /etc/firmware/
+FwPatchFileRootPath = /etc/firmware/bt/
diff --git a/libbt/src/conf.c b/libbt/src/conf.c
index 438f114..96f259d 100644
--- a/libbt/src/conf.c
+++ b/libbt/src/conf.c
@@ -36,6 +36,7 @@
 ******************************************************************************/
 int userial_set_port(char *p_conf_name, char *p_conf_value, int param);
 int hw_set_patch_file_path(char *p_conf_name, char *p_conf_value, int param);
+int hw_set_patch_file_root_path(char *p_conf_name, char *p_conf_value, int param); // Added by intel to be able to handle board revision
 int hw_set_patch_file_name(char *p_conf_name, char *p_conf_value, int param);
 #if (VENDOR_LIB_RUNTIME_TUNING_ENABLED == TRUE)
 int hw_set_patch_settlement_delay(char *p_conf_name, char *p_conf_value, int param);
@@ -73,6 +74,7 @@ typedef struct {
 static const conf_entry_t conf_table[] = {
     {"UartPort", userial_set_port, 0},
     {"FwPatchFilePath", hw_set_patch_file_path, 0},
+    {"FwPatchFileRootPath", hw_set_patch_file_root_path, 0}, // Added by intel to be able to handle board revision
     {"FwPatchFileName", hw_set_patch_file_name, 0},
 #if (VENDOR_LIB_RUNTIME_TUNING_ENABLED == TRUE)
     {"FwPatchSettlementDelay", hw_set_patch_settlement_delay, 0},
diff --git a/libbt/src/hardware.c b/libbt/src/hardware.c
index 98b4eba..ba970f1 100644
--- a/libbt/src/hardware.c
+++ b/libbt/src/hardware.c
@@ -1444,6 +1444,36 @@ int hw_set_patch_file_path(char *p_conf_name, char *p_conf_value, int param)
 
 /*******************************************************************************
 **
+** Function        hw_set_patch_file_root_path
+**
+** Description     Set the root location of firmware patch files. Then, the
+**                 board revision (e.g. "vv", "pr") etc is appended to this.
+**                 (created by intel)
+**
+** Returns         0
+**
+*******************************************************************************/
+int hw_set_patch_file_root_path(char *p_conf_name, char *p_conf_value, int param)
+{
+    // Buffer to get the value of the property giving the board revision
+    char board_revision[PROPERTY_VALUE_MAX] = {0};
+
+
+    /* First, copy the root file path received as parameter */
+    strcpy(fw_patchfile_path, p_conf_value); // as in hw_set_patch_file_path()
+
+    /* Then get and  append the board revision (e.g. vv, pr etc) */
+    property_get("ro.spid.wifi.nvram", // TODO: rename the property giving the board revision
+                 board_revision,
+                 "vv" /* if no property is set, assume it is a vv board */);
+    strcat(fw_patchfile_path, board_revision);
+    /* no need to append a final slash: hw_config_findpatch() copes with this. */
+
+    return 0;
+}
+
+/*******************************************************************************
+**
 ** Function        hw_set_patch_file_name
 **
 ** Description     Give the specific firmware patch filename
-- 
1.7.4.1

